torch/csrc/api/include/torch/data/dataloader.h

```diff
@@ -155,6 +166,13 @@ class DataLoader {
     sampler_.reset();
     sequence_number_ = 0;
     sequencer_ = new_sequencer();
+    if (main_thread_dataset_) {
+      main_thread_dataset_->reset();
+    } else {
+      for (size_t w = 0; w < workers_.size(); ++w) {
+          // How to communicate with a particular worker?
```

Comment:
We'll have to synchronize access to the dataset copies then, which is a bit annoying (effectively would need a list of datasets, and a list of mutexes, or maybe just one global mutex). It may be nicer to send workers "Reset" messages.

Reply:
We are calling reset() at the iterator begin, by that time the shuttle is drained already. Just wondering why we need to synchronize, we can let the synchronization to be handled by the dataset. For regular stateless datasets, they can be as it is without any impact.